test/e2e/apps/job.go

```diff
@@ -344,6 +344,57 @@ var _ = SIGDescribe("Job", func() {
 		}
 	})
 
+	ginkgo.It("should recreate pods only after they have failed if pod replacement policy is set to Failed", func(ctx context.Context) {
+		ginkgo.By("Creating a job")
+		job := e2ejob.NewTestJob("notTerminate", "pod-recreate-failed", v1.RestartPolicyNever, 1, 1, nil, 1)
+		job.Spec.Template.Finalizers = []string{"fake.example.com/blockDeletion"}
```

Comment:
the job pods already have finalizers, I don't think this additional finalizer would make a difference.

Reply:
yep, it already has the `job-tracking` finalizer, it is not necessary to add another one